mnist/README.md

```diff
@@ -6,165 +6,96 @@ This example guides you through the process of taking an example model, modifyin
+## 3.Kubeflow environment
 
 ```
-$ kubectl config set-credentials <username> --username=<admin_username> --password=<admin_password>
-$ kubectl config set-context <context_name> --cluster=<cluster_name> --user=<username> --namespace=<namespace>
-$ kubectl config use-context <context_name>
+git clone -b v0.3.2 https://github.com/kubeflow/kubeflow
+./kubeflow/scripts/kfctl.sh init kf-app --platform none
+cd kf-app
+../kubeflow/scripts/kfctl.sh generate k8s
+../kubeflow/scripts/kfctl.sh apply k8s
 ```
 
-### Local Setup
+For detail info, check here: https://www.kubeflow.org/docs/started/getting-started/
 
-You also need the following command line tools:
+Note: The vizier-db in Kubeflow depends on a PV, you need to create it in order to make vizier-db work. But this does not affect this example.
 
-- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
-- [argo](https://github.com/argoproj/argo/blob/master/demo.md#1-download-argo)
-- [ksonnet](https://ksonnet.io/#get-started)
 
-To run the client at the end of the example, you must have [requirements.txt](requirements.txt) intalled in your active python environment.
+## 4.Setup Minio
 
-```
-pip install -r requirements.txt
-```
+Minio provides an S3 compatible API, we will use Minio in this example to act as S3.
 
-NOTE: These instructions rely on Github, and may cause issues if behind a firewall with many Github users. Make sure you [generate and export this token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/):
+### Deploy Minio
 
 ```
-export GITHUB_TOKEN=xxxxxxxx
+docker run -e MINIO_ACCESS_KEY=minio -e MINIO_SECRET_KEY=minio123 \
+    --name=minio -d --net=host -v /var/lib/minio:/data minio/minio server /data
 ```
 
-## Modifying existing examples
-
-Many examples online use models that are unconfigurable, or don't work well in distributed mode. We will modify one of these [examples](https://github.com/tensorflow/tensorflow/blob/9a24e8acfcd8c9046e1abaac9dbf5e146186f4c2/tensorflow/examples/learn/mnist.py) to be better suited for distributed training and model serving.
-
-### Prepare model
-
-There is a delta between existing distributed mnist examples and what's needed to run well as a TFJob.
-
-Basically, we must:
+The Minio access key is `minio`, secret key is `minio123`. We will use them in later steps.
 
-1. Add options in order to make the model configurable.
-1. Use `tf.estimator.train_and_evaluate` to enable model exporting and serving.
-1. Define serving signatures for model serving.
 
-The resulting model is [model.py](model.py).
-
-### Build and push model image.
-
-With our code ready, we will now build/push the docker image.
+### Create Minio bucket
 
 ```
-DOCKER_BASE_URL=docker.io/elsonrodriguez # Put your docker registry here
```

Comment:
Removing this step makes the origin of the model container opaque, as stated by Jeremy in one of his reviews.

Reply:
There is already a dockerfile directory inside this example, and the steps are located there.

Later when I move the dockerfile to the location @jlewi suggested, I will update it to reference the link.